aiocoap/transports/udp6.py

```diff
@@ -247,11 +247,20 @@ def _local_port(self):
         return self.transport.get_extra_info('socket').getsockname()[1]
 
     @classmethod
-    async def _create_transport_endpoint(cls, sock, ctx: interfaces.MessageManager, log, loop, multicast=[]):
+    async def _create_transport_endpoint(cls, sock, ctx: interfaces.MessageManager, log, loop, multicast=[], broadcast: bool = False):
         try:
             sock.setsockopt(socket.IPPROTO_IPV6, socknumbers.IPV6_RECVPKTINFO, 1)
         except NameError:
             raise RuntimeError("RFC3542 PKTINFO flags are unavailable, unable to create a udp6 transport.")
+        if broadcast is True:
+            try:
+                # This shouldn't be needed, at least not when using --broadcast
+                # with aiocoap-client.
+                if not sock.getsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST):
+                    sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+            except Exception:
+                log.fatal("Unable to set socket to SO_BROADCAST; setsockopt() failed")
+                raise
```

Comment:
```suggestion
                log.warning("Unable to set socket to SO_BROADCAST; continuing, but requests can not be sent to broadcast addresses.")
```

When there is no extra flag, this can best-effort.